### DISCUSSION TITLE
Hard refresh when clicking on any <Link /> - on Vercel only

### DISCUSSION THREAD

**Discussion by cibulka** [vercel/community#1889] [Help] [answered]
### Summary

Hi everyone!

I've deployed my project to Vercel only to find out, that clicking on any `<Link />` component leads to a hard refresh of the page. This happens only on Vercel - everything is fine locally, both in development (`npm run dev`) and production (`npm run build`). 

Here's the link: https://dotu-web-mdx.vercel.app

There are also no errors anywhere - neither in the browser console, nor in server logs.

I'm using Next.js 13 with the app folder. I also need to use i18n, which unfortunately _still has not been standardized_ for Next.js 13. I've followed [this guide](https://beta.nextjs.org/docs/guides/internationalization) for the app folder setup and [this guide](https://locize.com/blog/next-13-app-dir-i18n/) to setup `i18next`. 

For the i18n reasons I'm using a wrapper around `<Link />` that adds the locale to href, if not present already. To be sure that this is not the culprit I've created a separate branch without the wrapper only to see that the problem persists: https://dotu-web-mdx-git-feat-links-without-locale-dotu.vercel.app

I've already texted Vercel support, but thought it might be useful to create a post as here, as I need to resolve the issue ASAP. I also understand that I should probably provide more info, but not sure what exactly (as there are really no errors whatsover). What would be helpful?

Thank you in advance and wish you coding without accidents :), P!

### Edit

Okay, now I have the reproductions. TLDR; It seems that the issue lies with Next.js 13.1.6 (to which I'm unfortunately locked, because of [contentlayer bug](https://github.com/contentlayerdev/contentlayer/issues/386)).

**The repo**: https://github.com/cibulka/next-i18n-test

You can observe the effect by clicking on *any* link. The hard refresh happens only **on Vercel**, locally (both in development and production) everything is fine.

| What | Error | Branch | Preview |
| --- | --- | --- | --- |
| Next 13.1.6 + Contentlayer | 🐛 | [main](https://github.com/cibulka/next-i18n-test/tree/main) | [link](https://next-i18n-test-dotu.vercel.app) |
| Next 13.1.6 + No Contentlayer | 🐛 | [no-contentlayer](https://github.com/cibulka/next-i18n-test/tree/no-contentlayer) | [link](https://next-i18n-test-git-no-contentlayer-cibulka.vercel.app) |
| Next 13.2.4 + No Contentlayer | 👍 | [no-contentlayer-13.2.4](https://github.com/cibulka/next-i18n-test/tree/no-contentlayer-13.2.4) | [link](https://next-i18n-test-git-no-contentlayer-cibulka.vercel.app) |

I suppose that if there is no problem on the current version of Next.js I can simply wait for the contentlayer compatibility update. I'm still curious why this is happening only on Vercel and without any error message visible.

### Example

https://dotu-web-mdx.vercel.app

**Comment by amyegan** [comment]
Thanks for sharing the example links, @cibulka! I see what you're talking about, but I can't tell why this is happening based solely on the example deployments. Are you able to share your repo or a minimal reproduction?

**Reply by cibulka** [reply]
Hey @amyegan, thanks for you reply! I can't share the repo, but during today I will be making a minimal reproduction and when I'm done, I'll share it here.

**Reply by cibulka** [reply]
Hey @amyegan, provided reproductions. Could you take a look please?

**Reply by amyegan** [reply, marked as answer]
Thanks for sharing these reproductions. The [App Router](https://beta.nextjs.org/docs/getting-started#introducing-the-app-router) is changing pretty rapidly in beta. Since [v13.1.6](https://github.com/vercel/next.js/releases/tag/v13.1.6) is a couple of months old, there have been a lot of changes since it's release. I don't know exactly what has changed since then to impact this full page refresh behavior, but it does look like 13.2 fixes it.

I hate to have to tell you that the solution is to wait for Contentlayer to support the latest Next.js version, but I think that's the way forward.  😞

**Comment by hanselabreu** [comment]
What version of `next` do you have? Bumping the version to the latest fixed it for me.

**Reply by cibulka** [reply]
The project uses [contentlayer](https://www.contentlayer.dev/) and because of [https://github.com/contentlayerdev/contentlayer/issues/386](this issue) I'm locked with 13.1.6. During today I'll be making a minimal reproduction and through this process I'll see if that's the issue. I'll post my results here. :)

**Reply by cibulka** [reply]
Hey @hanselabreu, it seems you were right! Out of curiosity - were you also using contentlayer or is this unrelated?

**Reply by hanselabreu** [reply]
I was not 🙂
